gkr_iop/src/selector.rs

```diff
@@ -26,8 +24,13 @@ pub enum SelectorType<E: ExtensionField> {
     Whole(Expression<E>),
     /// Select a prefix as the instances, padded with a field element.
     Prefix(E::BaseField, Expression<E>),
-    /// Select a prefix as the instances, for each instance select the 1st or the last round, padded with a field element.
-    KeccakRound(usize, E::BaseField, Expression<E>),
```

Comment:
I just figured out, even we try to introduce specific KeccakRound to reduce 5 num of variables, but in tower prover, the maximal number of variables still dominated by lookup, and lookup still + 5 num of variable.  Thus, the original optmisation on tower argument doesnt work, because right now we batched all layers and assume all with same height. And the end all share the same opening point.

Thus, we need to keep read/write record size to align with lookup. With that, an generalized `OrderedSparse32` was introduced